<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="mode" default="$(env MOVEL_MODE)" doc="mode: [deployment, development]"/>
  <arg name="rgb_color_topic" default="/camera/rgb/image_raw" />
  <arg name="rgbd_depth_topic" default="/camera/depth_registered/image_raw" />
  <arg name="rgbd_camera_info" default="/camera/rgb/camera_info" />

  <group if="$(eval mode == 'deployment')">
    <node name="orb_slam2_rgbd" pkg="orb_slam2_ros" type="orb_slam2_ros_rgbd" output="screen">
        <rosparam file="$(find orbomator)/config/rgbd_map.yaml"/>
        <rosparam file="$(env HOME)/.config/movel/config/orbomator/config/rgbd_map.yaml"/>
        <remap from="/camera/rgb/image_raw" to="$(arg rgb_color_topic)" />
        <remap from="/camera/depth_registered/image_raw" to="$(arg rgbd_depth_topic)" />
        <remap from="/camera/rgb/camera_info" to="$(arg rgbd_camera_info)" />
        <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />
    </node>
  </group>

  <group unless="$(eval mode == 'deployment')">
    <node name="orb_slam2_rgbd" pkg="orb_slam2_ros" type="orb_slam2_ros_rgbd" output="screen">
        <rosparam file="$(find orbomator)/config/rgbd_map.yaml"/>
        <remap from="/camera/rgb/image_raw" to="$(arg rgb_color_topic)" />
        <remap from="/camera/depth_registered/image_raw" to="$(arg rgbd_depth_topic)" />
        <remap from="/camera/rgb/camera_info" to="$(arg rgbd_camera_info)" />
        <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />
    </node>
  </group>